### PR TITLE
269 Renamed files and made an index page directory for admin pages.

### PR DESCRIPTION
**Pull request by cedricabuso**

### CODE REVIEW

File: admin/url_logs.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+            if (session) {
+                window.location.href = '#';
+                document.cookie = 'hbeat_access_token=' + session;
+            }
+
+            session = document.cookie.split('; ');
+
+            _(session)
+                .forEach(function (e) {
```
Comment (nhvu1988) [inline admin/url_logs.js]: should use _.find
```js
var session = _.find(sessions, function (e) {
    return ~e.indexOf('hbeat_access_token=');
});

session = session.split('hbeat_access_token=')[1];
```

File: admin/url_logs.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+        is_signed_in = function () {
+            $('.tabs').show();
+            
+            React.render(
+                <htbt.admin.Loader />,
+                $('#url-logs .match-container')[0]
+            );
+
+            session = window.location.href.split('#access_token=')[1];
```
Comment (nhvu1988) [inline admin/url_logs.js]: please declare exacly what variable is using for 
`var access_token = ...`

File: admin/url_logs.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+            if (session) {
+                window.location.href = '#';
+                document.cookie = 'hbeat_access_token=' + session;
+            }
+
+            session = document.cookie.split('; ');
```
Comment (nhvu1988) [inline admin/url_logs.js]: also here, please use s prefix if the variable is plural

```js
var cookies;
...
cookies = document.cookie.split('; ');
```

File: admin/url_logs.js [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+            _(session)
+                .forEach(function (e) {
+                    if (e.indexOf('hbeat_access_token=') > -1) {
+                        session = e.split('hbeat_access_token=')[1];
+                        return;
```
Comment (nhvu1988) [inline admin/url_logs.js]: `return` is not helpful here

File: admin/url_logs.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+            $.ajax({
+                type: 'GET',
+                url: 'http://api.accounts.freedom.tm/user',
+
+                headers: {'ACCESS-TOKEN': session},
+
+                success: is_admin,
+
+                error: function () {}
```
Comment (nhvu1988) [inline admin/url_logs.js]: can use _.noop

File: admin/url_logs.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+                );
+
+                $('#url-logs .center-align')[0].style.display = 'none';
+                return;
+            }
+
+            $.ajax({
+                type: 'GET',
+                url: htbt.config.backend + '/crm/user',
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+                data: {session: session},
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+                success: set_user,
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+                        <htbt.admin.Login />,
+                        $('#login-cont')[0]
+                    );
+                }
+            });
+
+            $.ajax({
+                type: 'GET',
+                url: 'http://api.accounts.freedom.tm/user',
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+                headers: {'ACCESS-TOKEN': session},
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+                success: is_admin,
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+                });
+        },
+
+        is_admin = function (e) {
+            email = e.email;
+
+            $.ajax({
+                type: 'GET',
+                url: htbt.config.backend + '/admin',
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 13]
```diff
@@ -0,0 +1,243 @@
...
+                <htbt.admin.Logout />,
+                $('#login-cont')[0]
+            );
+
+            $('#logout')
+                .click(function () {
+                    $.ajax({
+                        type: 'POST',
+                        url: 'http://api.accounts.freedom.tm/auth/logout',
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 14]
```diff
@@ -0,0 +1,243 @@
...
+                        headers: {'ACCESS-TOKEN': session},
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 15]
```diff
@@ -0,0 +1,243 @@
...
+        is_admin = function (e) {
+            email = e.email;
+
+            $.ajax({
+                type: 'GET',
+                url: htbt.config.backend + '/admin',
+
+                data: {email: e.email},
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 16]
```diff
@@ -0,0 +1,243 @@
...
+                success:get_passkey,
```
Comment (nhvu1988) [inline admin/url_logs.js]: space after `:`

File: admin/url_logs.js [hunk 17]
```diff
@@ -0,0 +1,243 @@
...
+
```
Comment (nhvu1988) [inline admin/url_logs.js]: remove empty line

File: admin/url_logs.js [hunk 18]
```diff
@@ -0,0 +1,243 @@
...
+        get_passkey = function () {
+            function get_access (e) {
+                var key = $('#access_code')[0].value;
+                
+                if (!key || key === ' ') {
+                    return;
+                }
+
+                if (e) {
```
Comment (nhvu1988) [inline admin/url_logs.js]: e && e.preventDefault();

File: admin/url_logs.js [hunk 19]
```diff
@@ -0,0 +1,243 @@
...
+                    );
+                }
+            });
+        },
+
+        get_passkey = function () {
+            function get_access (e) {
+                var key = $('#access_code')[0].value;
+                
+                if (!key || key === ' ') {
```
Comment (nhvu1988) [inline admin/url_logs.js]: should use `trim()`

**Comment by cedricabuso** [comment]
@nhvu1988 Updated.

File: admin/url_logs.js [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+
+            access_token = access_token.split('hbeat_access_token=')[1];
+
+            if (typeof(access_token) !== 'string') {
+                React.render(
+                    <htbt.admin.Login />,
+                    $('#login-cont')[0]
+                );
+
+                $('#url-logs .center-align')[0].style.display = 'none';
```
Comment (nhvu1988) [inline admin/url_logs.js]:
```js
$('#url-logs .center-align:eq(0)').hide();
```

File: admin/url_logs.js [hunk 21]
```diff
@@ -0,0 +1,224 @@
...
+                    );
+                }
+            });
+
+            $.ajax({
+                type: 'GET',
+                url: 'http://api.accounts.freedom.tm/user',
+                headers: {'ACCESS-TOKEN': access_token},
+                success: is_admin,
+                error: _.noop()
```
Comment (nhvu1988) [inline admin/url_logs.js]: just as below :)
```js
error: _.noop
```

File: admin/url_logs.js [hunk 22]
```diff
@@ -0,0 +1,224 @@
...
+                    );
+                }
+            });
+        },
+
+        get_passkey = function () {
+            function get_access (e) {
+                var key = $('#access_code')[0].value;
+                
+                if (!key || !key.trim()) {
```
Comment (nhvu1988) [inline admin/url_logs.js]: just as below 
```js
if (!key.trim()) 
```

File: admin/url_logs.js [hunk 23]
```diff
@@ -0,0 +1,224 @@
...
+                    return;
+                }
+
+                e && e.preventDefault();
+
+                $('#modal1').closeModal();
```
Comment (nhvu1988) [inline admin/url_logs.js]: what's `modal1` use for? and can you rename `modal1` to specific name?

File: admin/url_logs.js [hunk 24]
```diff
@@ -0,0 +1,224 @@
...
+                    error: function () {
+                        React.render(
+                            <htbt.admin.Error data='Access Denied' />,
+                            $('#url-logs .match-container')[0]
+                        );
+                    }
+                });
+            }
+
+            if (window.location.href.indexOf('/admin/#category') > -1) {
```
Comment (nhvu1988) [inline admin/url_logs.js]:
```js
if (~window.location.href.indexOf('/admin/#category')) {
```

**Comment by cedricabuso** [comment]
@nhvu1988 Updated :)

File: admin/url_logs.js [hunk 25]
```diff
@@ -0,0 +1,224 @@
...
+
+            access_token = document.cookie.split('; ');
+
+            access_token = _.find(access_token, function (e) {
+                return ~e.indexOf('hbeat_access_token=');
+            });
+
+            access_token = access_token.split('hbeat_access_token=')[1];
+
+            if (typeof(access_token) !== 'string') {
```
Comment (nhvu1988) [inline admin/url_logs.js]: what you want to check here? when this condition is true? how a value returned by `split` is not a string?

File: admin/admin.js [hunk 26]
```diff
@@ -398,7 +357,7 @@
                     $('#login-cont')[0]
                 );
 
-                $('#url-logs .center-align')[0].style.display = 'none';
+                $('#lfg-categories')[0].style.display = 'none';
```
Comment (nhvu1988) [inline admin/admin.js]: $('#lfg-categories').hide()

File: admin/url_logs.js (hunk 25 above)
Comment (cedricabuso) [inline admin/url_logs.js]: @nhvu1988 Ahh, that's an old code, did it for checking before, forgot to remove it. :)

File: admin/url_logs.js [hunk 27]
```diff
@@ -0,0 +1,224 @@
...
+                    );
+                }
+            });
+
+            $.ajax({
+                type: 'GET',
+                url: 'http://api.accounts.freedom.tm/user',
+                headers: {'ACCESS-TOKEN': access_token},
+                success: is_admin,
+                error: _.noop
```
Comment (nhvu1988) [inline admin/url_logs.js]: ah... sorry but you should remove this line, `error` callback is optional in $.ajax function :)

Comment (cedricabuso) [inline admin/url_logs.js]: @nhvu1988 ok :)

**Comment by cedricabuso** [comment]
@nhvu1988 Updated. :)

**Comment by nhvu1988** [comment]
@cedricabuso can we move 3 buttons to the page header for admin can switch pages easily?

image

**Comment by cedricabuso** [comment]
@nhvu1988 Move? As in remove the buttons on the page content and put them on the nav header?

**Comment by nhvu1988** [comment]
@cedricabuso yes :)

**Comment by cedricabuso** [comment]
ok :) Will change the lay out of the nav headers on the pages first. :)

**Comment by cedricabuso** [comment]
@nhvu1988 Updated. :)

**Pull request merged**